packages/core/lib/console.js

```diff
@@ -61,22 +61,31 @@ class Console extends EventEmitter {
 
   async start() {
     try {
-      // we calculate the context before calling `start`  so that the context
-      // variables are available as early as possible after the repl starts
-      const context = { ...(await this.calculateTruffleAndUserGlobals()) };
-      // we set up this.repl here because `this.provision` sets values on
-      // this.repl.context - this will finish setting up the environment
-      this.repl = { context };
-      // this hydrates the environment with the user's contracts
-      this.provision();
-
+      // start the repl with an empty prompt and show a proper one when
+      // the repl has set up its context and is ready to accept input
       this.repl = repl.start({
-        prompt: "truffle(" + this.options.network + ")> ",
+        prompt: "",
+        ignoreUndefined: true,
```

Comment:
Is this change considered breaking?

Reply:
I'm not sure, but leaning towards removing it.

Reply:
[Adding documentation for reviewers]
[ignoreUndefined](https://nodejs.org/dist/latest-v14.x/docs/api/repl.html#repl_repl_start_options) :- <boolean> If true, specifies that the default writer will not output the return value of a command if it evaluates to undefined. **Default**: false.

Reply:
Do we want this? It should behave like a normal Node repl, right?

Reply:
Removing it from this PR to be revisited in follow on work